mysql/src/finagle_clojure/mysql/client.clj

```diff
@@ -0,0 +1,346 @@
+(ns finagle-clojure.mysql.client
+  (:import (com.twitter.finagle Stack$Param)
+           (com.twitter.finagle.exp.mysql Row Client PreparedStatement Result OK ResultSet Field)
+           (com.twitter.finagle.exp Mysql$Client Mysql)
+           (com.twitter.util Future))
+  (:require [finagle-clojure.scala :as scala]
+            [finagle-clojure.mysql.value :as value]
+            [finagle-clojure.futures :as f]))
+
+(defn- Field->keyword [^Field f]
+  (-> f (.name) (keyword)))
+
+(defn Row->map
+  "Given a Finagle MySQL row structure, convert it to a Clojure map of field names to idiomatic
+  Java data structures. Raises an exception when it can't find an appropriate way to convert
+  the boxed MySQL value.
+
+  *Arguments:*
+
+    * `row`: a [[com.twitter.finagle.exp.mysql.Row]]
+
+  *Returns:*
+
+    a Clojure map of field/value pairs"
+  [^Row row]
+  (zipmap
+    (map Field->keyword (scala/scala-seq->vec (.fields row)))
+    (map value/unbox (scala/scala-seq->vec (.values row)))))
+
+(defn ResultSet->vec [^ResultSet rs]
+  "Given a Finagle MySQL result set, convert it to a vector of Clojure maps using `Row->map`. Raises
+  an exception when it can't find an appropriate way to convert the boxed MySQL value.
+
+  *Arguments:*
+
+    * `rs`: a [[com.twitter.finagle.exp.mysql.ResultSet]]
+
+  *Returns:*
+
+    a Clojure vector of maps"
+  (->> (.rows rs)
+       (scala/scala-seq->vec)
+       (map Row->map)))
```

Comment:
changed this to use `mapv` and return a vector in https://github.com/finagle/finagle-clojure/commit/f6a897137ccba42ec90fc21e737fab694b860900